subsys/kafka/src/main/java/org/commonjava/indy/subsys/kafka/handler/RepoServiceEventHandler.java

```diff
@@ -0,0 +1,356 @@
+                final ArtifactStore[] stores = storeMap.values().toArray( new ArtifactStore[storeMap.size()] );
+                final EventMetadata eventMetadata = convertEventMetadata( storeEvent );
+                Map<ArtifactStore, ArtifactStore> changeMap;
+                switch ( storeEvent.getEventType() )
+                {
+                    case PreDelete:
+                        dispatcher.deleting( eventMetadata, stores );
+                        break;
+                    case PostDelete:
+                        dispatcher.deleted( eventMetadata, stores );
+                        break;
+                    case PreUpdate:
+                        StorePreUpdateEvent preUpdateEvent = mapper.readValue( value, StorePreUpdateEvent.class );
+                        changeMap = getChangeMap( preUpdateEvent, storeMap );
+                        dispatcher.updating( ArtifactStoreUpdateType.valueOf( preUpdateEvent.getUpdateType().name() ),
+                                             eventMetadata, changeMap );
+                        break;
+                    case PostUpdate:
+                        StorePostUpdateEvent postUpdateEvent = mapper.readValue( value, StorePostUpdateEvent.class );
+                        changeMap = getChangeMap( postUpdateEvent, storeMap );
+                        dispatcher.updating( ArtifactStoreUpdateType.valueOf( postUpdateEvent.getUpdateType().name() ),
+                                             eventMetadata, changeMap );
+                        break;
+                    case Enablement:
+                        StoreEnablementEvent enablementEvent = mapper.readValue( value, StoreEnablementEvent.class );
+                        boolean disabling = enablementEvent.isDisabling();
+                        boolean preprocessing = enablementEvent.isPreprocessing();
+                        if (!disabling && preprocessing)
+                        {
+                            dispatcher.enabling( eventMetadata, stores );
+                        }
+                        if ( !disabling && !preprocessing )
+                        {
+                            dispatcher.enabled( eventMetadata, stores );
+                        }
+                        if ( disabling && preprocessing )
+                        {
+                            dispatcher.disabling( eventMetadata, stores );
+                        }
+                        if ( disabling && !preprocessing )
+                        {
+                            dispatcher.disabled( eventMetadata, stores );
+                        }
+                        break;
+                }
+                logger.info( "Finish the consumer event dispatcher for event type {}",
+                              storeEvent.getEventType().name() );
+            }
+            catch ( JsonProcessingException e )
+            {
+                logger.error( String.format(
```

Comment:
Maybe we should consider these error handling strategy later to use some alternative way, like an error handling queue to store these events. Or store them into cassandra?

Reply:
@ligangty May I know what's the purpose to store events handling errors? Maybe we can add it into the next cycle for event handler WP implementation

Reply:
If the event processing encountered errors and just logged, we may missed the reasons for these events because of volatile log messages. I'm still thinking the strategy to handle these stuff, like persistence. I agree to give implement them in a separate WP